src/main/java/com/aws/greengrass/detector/IpDetectorService.java

```diff
@@ -5,51 +5,62 @@
 
 package com.aws.greengrass.detector;
 
+import com.aws.greengrass.componentmanager.KernelConfigResolver;
+import com.aws.greengrass.config.Topic;
 import com.aws.greengrass.config.Topics;
 import com.aws.greengrass.dependency.ImplementsService;
 import com.aws.greengrass.dependency.State;
 import com.aws.greengrass.detector.config.Config;
+import com.aws.greengrass.lifecyclemanager.Kernel;
 import com.aws.greengrass.lifecyclemanager.PluginService;
+import com.aws.greengrass.lifecyclemanager.exceptions.ServiceLoadException;
+import com.aws.greengrass.util.Coerce;
 
 import java.util.concurrent.Future;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
 import javax.inject.Inject;
 
 
-@ImplementsService(name = IpDetectorService.DECTECTOR_SERVICE_NAME)
+@ImplementsService(name = IpDetectorService.DETECTOR_SERVICE_NAME)
 public class IpDetectorService extends PluginService {
-    public static final String DECTECTOR_SERVICE_NAME = "aws.greengrass.clientdevices.IPDetector";
+    public static final String DETECTOR_SERVICE_NAME = "aws.greengrass.clientdevices.IPDetector";
+    static final String MOQUETTE_SERVICE_NAME = "aws.greengrass.clientdevices.mqtt.Moquette";
+    static final String MOQUETTE = "moquette";
+    static final String PORT = "ssl_port";
```

Comment:
I want to know what happens when the port=`0` provided by the customer for the ssl_port config to moquette. Usually it means to select a random port. In that case, this approach will not work and we will need moquette to publish the resolved port.

Reply:
That is correct. If we are supporting the port 0 as random case, we need to update the moquette code for that case. Also in all our future brokers if we are supporting this.